src/index.tsx

```diff
@@ -663,8 +664,26 @@ class H5AudioPlayer extends Component<PlayerProps> {
   }
 
   componentDidUpdate(prevProps: PlayerProps): void {
-    const { src, autoPlayAfterSrcChange } = this.props
+    const { src, autoPlayAfterSrcChange, children } = this.props
+
+    let isAudioSrcChanged = false
+
     if (prevProps.src !== src) {
+      isAudioSrcChanged = true
+    } else if (children && Children.count(children) > 0) {
```

Comment:
probably not the best idea to check the children like this, but does add to the convenience for autoplay.

alternatively i could remove it, leaving the dev to implement autoplay when using `<source>` elements, perhaps an approach with `useEffect` with the playlist index changes.